### PR TITLE
feat: port `influxd inspect dump-tsm`

### LINKED ISSUE #19532: Port influxd inspect dump-tsm command
## What

Port the `influxd inspect dump-tsm` command from InfluxDB 1.x.

## How

Original 2.0 sub command:

None existed

Original 1.x source:

https://github.com/influxdata/influxdb/blob/42873d44244c23317b274221ffad8a065ae0bb0c/cmd/influx_inspect/dumptsm/dumptsm.go#L40

### PR DESCRIPTION
**Pull request by DStrand1**
Closes #19532

Ports the `./influx inspect dump-tsm` subcommand to 2.x.

- [ ] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [X] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [X] Rebased/mergeable
- [x] Tests pass

### CODE REVIEW

File: cmd/influxd/inspect/dump_tsm/dump_tsm.go [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+
+		fmt.Fprintln(tw, "  "+strings.Join([]string{"Pos", "Min Time", "Max Time", "Ofs", "Size", "Key", "Field"}, "\t"))
+		var pos int
+		for i := 0; i < keyCount; i++ {
+			key, _ := r.KeyAt(i)
+			for _, e := range r.Entries(key) {
+				pos++
+				split := strings.Split(string(key), "#!~#")
+
+				// Possible corruption? Try to read as much as we can and point to the problem.
```
Comment (danxmoran) [inline cmd/influxd/inspect/dump_tsm/dump_tsm.go:105]: Not sure what the "Possible corruption?" in this comment is referring to

File: cmd/influxd/inspect/dump_tsm/dump_tsm.go [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+		}
+	}
+
+	if args.dumpBlocks {
+		cmd.Println("Blocks:")
+		tw.Flush()
+		cmd.Println()
+	}
+
+	var blockSizeAvg int64
```
Comment (danxmoran) [inline cmd/influxd/inspect/dump_tsm/dump_tsm.go:211]: From here down feels like it should be under the `if args.dumpBlocks`

File: cmd/influxd/inspect/dump_tsm/dump_tsm.go [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+				}, "\t"))
+			}
+
+			i += blockSize
+			blockCount++
+		}
+	}
+
+	if args.dumpBlocks {
+		cmd.Println("Blocks:")
```
Comment (danxmoran) [inline cmd/influxd/inspect/dump_tsm/dump_tsm.go:206]: Can you move this logic (and anything else that moves under `if args.dumpBlocks`) into a helper function?

**Comment by DStrand1** [comment]
Should I mention the change to `dumpBlocks` and `dumpIndex` in the changelog or elsewhere?

File: cmd/influxd/inspect/dump_tsm/dump_tsm.go [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+				return nil
+			}
+			arguments.dumpBlocks = arguments.dumpBlocks || arguments.dumpAll
+			arguments.dumpIndex = arguments.dumpIndex || arguments.dumpAll
+			return dumpTSM(cmd, arguments)
+		},
+	}
+
+	cmd.Flags().StringVar(&arguments.path, "file", "",
+		"Root storage path")
```
Comment (danxmoran) [inline cmd/influxd/inspect/dump_tsm/dump_tsm.go:42]: We expect users to pass a TSM file as this arg, right? If so then I'd suggest:
```suggestion
		"Path to TSM file")
```

File: cmd/influxd/inspect/dump_tsm/dump_tsm.go [hunk 5]
```diff
@@ -0,0 +1,341 @@
...
+			tw:       tw,
+			minTime:  minTime,
+			maxTime:  maxTime,
+			keyCount: keyCount,
+			r:        r,
+		})
+	}
+
+	tw = tabwriter.NewWriter(cmd.OutOrStdout(), 8, 8, 1, '\t', 0)
+	fmt.Fprintln(tw, "  "+strings.Join([]string{"Blk", "Chk", "Ofs", "Len", "Type", "Min Time", "Points", "Enc [T/V]", "Len [T/V]"}, "\t"))
```
Comment (danxmoran) [inline cmd/influxd/inspect/dump_tsm/dump_tsm.go:102]: Should this print only run `if args.dumpBlocks`?

**Review by danxmoran — COMMENTED** [review]
A few final suggestions

**Review by danxmoran — APPROVED** [review]
LGTM in manual test

**Pull request merged**